packages/fxa-settings/src/components/DropDownAvatarMenu/index.tsx

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import Avatar from '../Avatar';
+import { useAccount } from '../../models';
+import { useClickOutsideEffect } from 'fxa-react/lib/hooks';
+import { useEscKeydownEffect } from '../../lib/hooks';
+import { ReactComponent as SignOut } from './sign-out.svg';
+
+export const DropDownAvatarMenu = () => {
+  const { displayName, primaryEmail } = useAccount();
+  const [isRevealed, setRevealed] = useState(false);
+  const toggleRevealed = () => setRevealed(!isRevealed);
+  const avatarMenuInsideRef = useClickOutsideEffect<HTMLDivElement>(
+    setRevealed
+  );
+  useEscKeydownEffect(setRevealed);
+  const dropDownId = 'drop-down-avatar-menu';
+
+  //TODO: mutation for Sign out
```

Comment:
might be nice to include the bug number in the TODO

Reply:
This was why my PR was still a draft. Pushing updates in a sec...